contracts/staking/validatorShare/ValidatorShare.sol

```diff
@@ -81,11 +82,12 @@ contract ValidatorShare is ValidatorShareStorage {
         logger.logStakeUpdate(validatorId);
     }
 
-    function sellVoucher() public {
+    function sellVoucher(uint256 expectedExchangeRate) public {
         uint256 share = balanceOf(msg.sender);
         require(share > 0, "Zero balance");
-        uint256 _amount = exchangeRate().mul(share).div(EXCHANGE_RATE_PRECISION);
-
+        uint256 rate = exchangeRate();
+        uint256 _amount = rate.mul(share).div(EXCHANGE_RATE_PRECISION);
+        require(rate >= expectedExchangeRate, "More slippage then expectedExchangeRate");
```

Comment:
We should move to `89`. We should do the calculation after the require condition.

Reply:
ah yes